### PR TITLE
Fix broken links

### PR DESCRIPTION
**Pull request by bhrutledge**
Discovered while looking at some open PR's, and currently failing on `main`:

https://github.com/pypa/packaging.python.org/runs/3850844441?check_suite_focus=true

### CODE REVIEW

File: source/key_projects.rst [hunk 1]
```diff
@@ -180,7 +180,7 @@ causing dependency conflicts with other packages installed on the system.
 Python Packaging User Guide
 ===========================
 
-`Docs <https://packaging.python.org/en/latest/>`__ |
+`Docs <https://packaging.python.org/>`__ |
```
Comment (bhrutledge) [inline source/key_projects.rst:183]: This one is surprising to me. Should https://packaging.python.org/en/latest/ work?

File: source/specifications/core-metadata.rst [hunk 2]
```diff
@@ -733,4 +733,4 @@ Examples::
 .. _`Python Package Index`: http://pypi.org/
 
 .. [2] RFC 822 Long Header Fields:
-   http://www.freesoft.org/CIE/RFC/822/7.htm
+   https://datatracker.ietf.org/doc/html/rfc822#section-3.1.1
```
Comment (webknjaz) [inline source/specifications/core-metadata.rst:736]: For RFCs, just use the `:rfc:` role in the source.

File: source/key_projects.rst (hunk 1 above)
Comment (webknjaz) [reply source/key_projects.rst:183]: Also, I thought it's been fixed in one of the other PRs. Let me see if it got merged...

**Review by webknjaz — COMMENTED** [review]
Merging #975, check for conflicts afterwards.

File: source/key_projects.rst (hunk 1 above)
Comment (meowmeowmeowcat) [reply source/key_projects.rst:183]: I've used intersphinx for this in #975:
```rst
:doc:`Docs <packaging.python.org:index>`
```

**Comment by bhrutledge** [comment]
I forgot about #975. Looks like this wasn't necessary. 🤦

**Comment by webknjaz** [comment]
Sorry I didn't merge that one sooner, it was updated while I was on vacation and I forgot about it for a bit...

**Pull request closed without merging**